internal/tools/aws/route53.go

```diff
@@ -47,15 +47,58 @@ func (a *Client) CreatePrivateCNAME(dnsName string, dnsEndpoints []string, logge
 
 // GetPrivateZoneDomainName gets the private Route53 domain name.
 func (a *Client) GetPrivateZoneDomainName(logger log.FieldLogger) (string, error) {
-	id, err := a.getHostedZoneIDWithTag(Tag{
+	id, err := a.GetPrivateZoneID(logger)
+	if err != nil {
+		return "", err
+	}
+	return a.getZoneDNS(id, logger)
+}
+
+// GetPrivateZoneID returns the R53 hosted zone ID for a given tag or if not provided for the default one
+func (a *Client) GetPrivateZoneID(logger log.FieldLogger, tags ...Tag) (string, error) {
+	if len(tags) > 1 {
```

Comment:
If this was changed to `if len(tags != 1)` then the conditional on line 66 could be removed.

Reply:
if we do that, function will return empty when we will have `len(tags) == 0` 🤔  right?

Reply:
Do we maybe just want to split this function up? One function for the default and one for tag(s)? It would probably make more sense when reading the code as well.

Reply:
@gabrieljackson I removed the rest of the code because currently is not used so I left the function as is for getting the default tags

Reply:
> if we do that, function will return empty when we will have len(tags) == 0 thinking right?

That doesn't seem right, if `len(tags)==0` then `tag` is kept as `tag := Tag{...` on line 62. That would be used in the query to `getHostedZoneIDWithTag`.

Reply:
> if we do that, function will return empty when we will have len(tags) == 0 thinking right?

No, if you provided an empty `tags` list, it would return the error on line 60.
I also made a typo in my suggestion, I meant to suggest `len(tags) != 1`.

Reply:
> we do that, function will return empty when we will have len(tags) == 0

It'll return `"", err`, on line 60. I can see how this would alter the default behavior, so I think you can disregard this comment, actually. I even made a typo in my original comment so either your code changed or I didn't fully understand the code when I left that comment.